hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOMDbCheckpointServlet.java

```diff
@@ -519,6 +524,18 @@ public void testWriteDbDataToStream() throws Exception {
         "expected snapshot files not found");
   }
 
+  private static long tmpHardLinkFileCount() throws IOException {
+    Path tmpDirPath = Paths.get(System.getProperty(JAVA_IO_TMPDIR));
```

Comment:
There is a danger with this method.  If this isn't the correct temp directory, then this code will always return 0 and the test will always pass.

Is it hard to add a way to be sure the file was actually created here?

Reply:
Tmp file is created using Java's NIO lib like `Files.createTempFile`. Tmp dir location is taken from static final field `java.nio.file.TempFileHelper.tmpdir` which uses `java.io.tmpdir` property value. That value is set in regards to which OS app is running on, but can be adjusted during runtime with `-Djava.io.tmpdir`. With all that said I think it is safe to leave it as is.

> Is it hard to add a way to be sure the file was actually created here?

I'd say this is not really the place for that. Such a test should be added to `TestOmSnapshotUtils` IMO since method that creates a file originates from `OmSnapshotUtils`. What do you think?